Peaky/Test.cs

```diff
@@ -18,5 +21,36 @@ internal class Test
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public Parameter[] Parameters { get; set; }
+
+        public static IEnumerable<Test> CreateTests(TestTarget testTarget, TestDefinition definition, HttpRequest request)
+        {
+            var testCases = testTarget.DependencyRegistry.GetParameterSetsFor(definition.TestMethod)?.ToList();
+
+            if (testCases != null && testCases.Count > 0)
+            {
+                foreach (var testCase in testCases)
+                {
+                    yield return new Test
+                    {
+                        Environment = testTarget.Environment,
+                        Application = testTarget.Application,
+                        Url = request.GetLinkWithQuery(testTarget, definition,testCase),
+                        Tags = definition.Tags,
+                        Parameters = testCase.ToArray()
+                    };
+                }
+            }
+
+            yield return new Test
+            {
+                Environment = testTarget.Environment,
+                Application = testTarget.Application,
+                Url = definition.Parameters.Any()? request.GetLinkWithQuery(testTarget, definition, definition.Parameters) : request.GetLink(testTarget, definition),
+                Tags = definition.Tags,
+                Parameters = definition.Parameters.Any()
```

Comment:
An empty set would be preferable to null. And with this many properties now on `Test`, it would be safer to shift to a constructor that guards and coalesces appropriately.

Reply:
Ok, I was being compliant with previous parameter handling where no parameters where converted to null collection